src/qt/atomicswapsdialog.cpp

```diff
@@ -0,0 +1,822 @@
+                ui->summaryLabel->setText(loadedSwap->GetSummary());
+                break;
+            }
+            case AtomicSwapType::Sell:
+            {
+                ui->tradeTypeLabel->setText(tr("Sell Order (You are purchasing assets)"));
+                ui->assetNameLabel->setText(formattedProvided);
+                ui->totalPriceLabel->setText(formattedExpected);
+                auto formattedUnit = RavenUnits::formatWithCustomName(loadedSwap->ExpectedType, loadedSwap->UnitPrice);
+                ui->unitPriceLabel->setText(QString("%1/[%2]").arg(formattedUnit).arg(loadedSwap->ProvidedType));
+                ui->summaryLabel->setText(loadedSwap->GetSummary());
+                break;
+            }
+            case AtomicSwapType::Trade:
+            {
+                ui->tradeTypeLabel->setText(tr("Trade Order (You are trading assets for assets)"));
+                ui->assetNameLabel->setText(formattedProvided);
+                ui->totalPriceLabel->setText(formattedExpected);
+                auto formattedUnit = RavenUnits::formatWithCustomName(loadedSwap->ProvidedType, loadedSwap->UnitPrice);
+                ui->unitPriceLabel->setText(QString("%1/[%2]").arg(formattedUnit).arg(loadedSwap->ExpectedType));
+                ui->summaryLabel->setText(loadedSwap->GetSummary());
+                break;
+            }
+        }
+
+        CMutableTransaction finalTX;
+        if(AttemptCompleteTransaction(finalTX, *loadedSwap, errorMessage))
+        {
+            this->validSwap = true;
+
+            loadedSwap.get()->SignedFinal = finalTX;
+
+            showValidMessage("Valid!");
+            enableExecuteButton();
+        }
+        else
+        {
+            this->validSwap = false;
+            
+            showMessage(tr("Unable to complete signed partial.\n%1").arg(errorMessage));
+            disableExecuteButton();
+        }
+    } else {
+        this->validSwap = false;
+        if(!errorMessage.isEmpty())
+            showMessage(tr("Invalid Swap.\n%1").arg(errorMessage));
+        disableExecuteButton();
+    }
+}
+
+bool AtomicSwapsDialog::AttemptParseTransaction(AtomicSwapDetails& result, QString& errorMessage)
+{
+    if (!DecodeHexTx(result.Transaction, result.Raw, true))
+    {
+        errorMessage = tr("Unable to decode partial transaction");
+        return false;
+    }
+        
+    if(result.Transaction.vin.size() != 1 || result.Transaction.vout.size() != 1)
+    {
+        errorMessage = tr("Input contains an invalid number of inputs/outputs.");
+        return false;
+    }
+    
+    CTxIn swap_vin = result.Transaction.vin[0];
+    CTxOut swap_vout = result.Transaction.vout[0];
+
+    auto vin_script = ScriptToAsmStr(swap_vin.scriptSig, true);
+    auto fIsSingleSign = vin_script.find("[SINGLE|ANYONECANPAY]") != std::string::npos;
+
+    if(!fIsSingleSign)
+    {
+        errorMessage = tr("Must be signed with SINGLE|ANYONECANPAY");
```

Comment:
Should probably split this, leave SINGLE|ANYONECANPAY out of tr().

Reply:
Ok, what would you think replacement error should be? "Signed partial not constructed properly." or something like that?

Reply:
The message can be the same, but don't translate "SINGEL|ANYONECANPAY"

Reply:
resolved in 993f991